compiler/lexer.nim

```diff
@@ -340,10 +340,15 @@ proc getNumber(L: var TLexer, result: var TToken) =
       else:
         break
       if buf[pos] == '_':
-        if buf[pos+1] notin chars:
+        if buf[pos+1] notin chars and buf[pos+1] == '_':
           lexMessage(L, errGenerated,
             "only single underscores may occur in a token: '__' is invalid")
           break
+        elif buf[pos+1] notin chars + {'_'}:
```

Comment:
Do we really need this special case here?

Reply:
I wasn't certain if a simple `else` would suffice. Of course if you mean the whole case, one could change the error message of the branch above to describe both double underscore + literal ending in underscore.
Personally I find it nicer to have the cases be separate, but if that's considered too verbose / not enough gain for the performance impact even (?), I'm fine with the combined error message. I don't have enough experience in the compiler to judge that.

Reply:
It's not about performance, this code is nowhere near the critical path. It's about effort vs benefit. Who writes ``1_`` and needs a super specialized error message for that?